File: java-algorithm/src/main/java/com/leetcode/Problem1470.java
package com.leetcode;

/**
 * 1470. Shuffle the Array
 * https://leetcode.com/problems/shuffle-the-array/
 */
public class Problem1470 {
    public int[] shuffle(int[] nums, int n) {
        int[] ans = new int[2 * n];
        int left = 0, right = n;
        for (int i = 0; i < ans.length; i++) {
            ans[i] = i % 2 == 0 ? nums[left++] : nums[right++];
        }
        return ans;
    }
}

File: python-algo/leetcode/problem_1470.py
"""1470. Shuffle the Array
https://leetcode.com/problems/shuffle-the-array/
"""
from typing import List


class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        left, right = [], []
        for i in range(n):
            left.append(nums[i])
            right.append(nums[i + n])
        for i in range(n):
            nums[i * 2] = left[i]
            nums[i * 2 + 1] = right[i]
        return nums
